R/as_gt.R

```diff
@@ -142,3 +142,16 @@ as_gt.simtrial_gs_wlr <- function(x,
       gt::tab_header(title = title, subtitle = subtitle)
   }
 }
+
+# Only purpose of the methods below is to fix S3 redirection when simtrial is
+# loaded after gsDesign2, which masks the as_gt() generic from gsDesign2
+
+#' @export
+as_gt.fixed_design <- function(x, ...) {
+  gsDesign2:::as_gt.fixed_design(x, ...)
```

Comment:
Regarding `:::`, actually I forgot to ask whether it would work by calling `gsDesign::as_gt()` instead of the specific method: 

```suggestion
  gsDesign2::as_gt(x, ...)
```

Of course, it will be better if we could avoid using `:::`.

Reply:
Unfortunately it doesn't work with only `::`

```R
library("gsDesign2")
library("simtrial")
## Attaching package: ‘simtrial’
##
## The following object is masked from ‘package:gsDesign2’:
##
##     as_gt
##

gs_design_ahr() |> summary() |> as_gt()
## Error: 'as_gt.gs_design' is not an exported object from 'namespace:gsDesign2'
```

Reply:
Interesting. Thanks for confirming!